feature/interface/singleton/otg_tests/singleton_test/singleton_test.go

```diff
@@ -371,11 +450,11 @@ func (tc *testCase) run(t *testing.T) {
 	for _, c := range []struct {
 		ipName     string
 		shouldFrag bool
-		ipHeader   ondatra.Header
+		ipHeader   string
```

Comment:
Also add the neighbor watcher as a wait function?

Here: `waitFn func(otg *otg.OTG)`

And provide separate functions for waitOTGIPv4 and waitOTGIPv6?

Reply:
I created separate functions for IPv4 and IPv6 but I don't understand the request for neighbor watcher as a wait function. Please take a look at the code and let me know

Reply:
switch removed and config and wait functions are now passed in the test run() function. The tests are now referring this functions in the struct